### PR TITLE
Training GenCast

### PR DESCRIPTION
**Pull request by gbruno16**
# Training Gencast

## Description

This PR is currently a draft and implements several modifications:

1. **Registering Graph Tensors as Buffers:**
   - The graph's tensors are registered as buffers during initialization. This allows them to move along with the model in a multi-device setting. 
   - Since `persistent=False`, these tensors will not be part of the `state_dict`.

2. **Wrapping the Model with PyTorch Lightning:**
   - The model is wrapped using PyTorch Lightning to facilitate easier training with various strategies (e.g., DDP, fp16).
   - This includes support for different schedulers (e.g., cosine annealing), loggers (e.g., wandb) and callbacks.
 
3. **Normalize first in sparse transformers:**
    - The flag `norm_first` in sparse transformers move the conditional layer normalization before the multi-head attention. This seems to improve training performance.
   
4. **Noise generation with torch_harmonics**
   - torch_harmonics supports noise generation on gpu with torch backend.

### CODE REVIEW

File: graph_weather/models/gencast/README.md [hunk 1]
```diff
@@ -125,3 +125,92 @@ preds = sampler.sample(denoiser, prev_inputs)
...
+    atmospheric_features=atmospheric_features,
+    single_features=single_features,
+    static_features=static_features,
+    max_year=2018,
+    time_step=2,
+)
+
+# download weights from HF
+print("> Downloading model's weights...")
+denoiser=Denoiser.from_pretrained("openclimatefix/gencast-128x64",
```
Comment (jacobbieker) [inline graph_weather/models/gencast/README.md]: I'd also add this as a test, to just run it to make sure it loads correctly.

File: tests/test_model.py [hunk 2]
```diff
@@ -466,48 +466,14 @@ def test_gencast_sampler():
 
 
 def test_gencast_full():
```
[the remark this answers is not in the document]
Comment (gbruno16) [reply tests/test_model.py]: Done!
